### ISSUE TITLE
Cannot `copy` a DataFrame containing a DictEncoded field with a missing value

### ISSUE THREAD

**Issue by yurivish** [JuliaData/Arrow.jl#101]
The `DictEncode` wrapper is the [recommended](https://discourse.julialang.org/t/ann-arrow-jl-0-3-release/47727/7) way to dict-encode a single column of a `DataFrame`.

I recently noticed that the presence of a `missing` value in a dict-encoded field seems to inhibit the ability to `copy` the `DataFrame`. Here is a minimal example:

```julia
julia> using Arrow, DataFrames

julia> let df = DataFrame(value = ["a", missing])
               df.value = Arrow.DictEncode(df.value)
               Arrow.write("df.arrow", df)
       end
"df.arrow"

julia> df = DataFrame(Arrow.Table("df.arrow"))
2×1 DataFrame
 Row │ value
     │ String?
─────┼─────────
   1 │ a
   2 │ missing

julia> copy(df)
ERROR: MethodError: Cannot `convert` an object of type Missing to an object of type Int8
Closest candidates are:
  convert(::Type{T}, ::T) where T<:Number at number.jl:6
  convert(::Type{T}, ::Number) where T<:Number at number.jl:7
  convert(::Type{T}, ::Ptr) where T<:Integer at pointer.jl:23
  ...
Stacktrace:
 [1] setindex!(::Array{Int8,1}, ::Missing, ::Int64) at ./array.jl:847
 [2] copy(::Arrow.DictEncoded{Union{Missing, String},Int8,SentinelArrays.ChainedVector{Union{Missing, String},Arrow.List{Union{Missing, String},Int32,Array{UInt8,1}}}}) at /Users/yurivish/.julia/packages/Arrow/CyJ4L/src/arraytypes/dictencoding.jl:209
 [3] DataFrame(::Array{AbstractArray{T,1} where T,1}, ::DataFrames.Index; copycols::Bool) at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/dataframe/dataframe.jl:195
 [4] manipulate(::DataFrame, ::Base.OneTo{Int64}; copycols::Bool, keeprows::Bool, renamecols::Bool) at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/abstractdataframe/selection.jl:1182
 [5] #manipulate#392 at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/abstractdataframe/selection.jl:1191 [inlined]
 [6] #select#384 at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/abstractdataframe/selection.jl:847 [inlined]
 [7] getindex at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/dataframe/dataframe.jl:473 [inlined]
 [8] #copy#183 at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/dataframe/dataframe.jl:844 [inlined]
 [9] copy(::DataFrame) at /Users/yurivish/.julia/packages/DataFrames/yqToF/src/dataframe/dataframe.jl:843
 [10] top-level scope at REPL[4]:1
```

**Comment by quinnj** [comment]
Please update to version 1.1, which was released yesterday, and included a fix for this.

**Comment by yurivish** [comment]
Thanks – just updated and can confirm the issue is fixed in 1.1.
